force-app/main/default/classes/SearchContactsonAccountExtension.cls

```diff
@@ -54,32 +60,19 @@ public with sharing class SearchContactsonAccountExtension {
             childContacts = Database.query(searchQuery);
         }
         catch(Exception ex){
-            ApexPages.Message myMsg = new ApexPages.Message(ApexPages.Severity.ERROR,string.valueof(ex)); 
-            ApexPages.addmessage(myMsg);
+            ApexPages.addmessage(new ApexPages.Message(ApexPages.Severity.ERROR,string.valueof(ex))); 
         }
     }
-    
+
     public void setAsPrimary(){
-        
+
         try{
-            //Reset contacts is primary attribute to false
-            List<Contact> actualContacts = new List<Contact>();
-      	    List<Contact> contactListToReset = new List<Contact>();
-      
-            for(Contact contact : [SELECT Id, Is_Primary_Contact__c FROM Contact WHERE AccountId = :accountRecord.Id AND (Is_Primary_Contact__c = true OR Id = :selectedContactId)]){
-                if(contact.Id != selectedContactId){
-                    contact.Is_Primary_Contact__c = false;                
-                }
-                else {
-                    contact.Is_Primary_Contact__c = true;
-                }
-                dddcontactListToReset.add(contact);
-            }
```

Comment:
This logic was correct. What I was trying to tell is to check if the contact is not already false. 
               if (contact.Id == selectedContactId) {
                    contact.Is_Primary_Contact__c = true;                
                }
                else if (contact.Is_Primary_Contact__c) {
                   // set Is_Primary_Contact__c  to false only for contacts which have Is_Primary_Contact__c true
                    contact.Is_Primary_Contact__c = false;
                }

Reply:
Ana, I think that this piece of code is outdated.
I removed this logic, because now we have a validation that two contacts cannot be primary on the sema account. So there is no need for reset.

Reply:
I see. It's good that you took into account the changes made for the trigger. 
But the trigger can be deactivated resulting in situations where we have 2 primary contacts for the same account.